bundles/org.openhab.binding.zwavejs/src/main/java/org/openhab/binding/zwavejs/internal/api/ZWaveJSClient.java

```diff
@@ -0,0 +1,388 @@
+            }
+        }, RECONNECT_INTERVAL_MINUTES, RECONNECT_INTERVAL_MINUTES, TimeUnit.MINUTES);
+    }
+
+    private void stopKeepAlive() {
+        ScheduledFuture<?> keepAliveFuture = this.keepAliveFuture;
+        if (keepAliveFuture != null) {
+            keepAliveFuture.cancel(true);
+        }
+        this.keepAliveFuture = null;
+    }
+
+    @Override
+    public void onWebSocketConnect(@NonNullByDefault({}) Session session) {
+        logger.debug("onWebSocketConnect('{}')", session);
+        this.session = session;
+        if (session != null) {
+            final WebSocketPolicy currentPolicy = session.getPolicy();
+            currentPolicy.setInputBufferSize(bufferSize);
+            currentPolicy.setMaxTextMessageSize(bufferSize);
+            currentPolicy.setMaxBinaryMessageSize(bufferSize);
+            keepAliveFuture = scheduler.scheduleWithFixedDelay(() -> {
+                try {
+                    String data = "Ping";
+                    ByteBuffer payload = ByteBuffer.wrap(data.getBytes());
+                    session.getRemote().sendPing(payload);
+                } catch (IOException e) {
+                    logger.warn("Problem starting periodic Ping. {}", e.getMessage());
+                }
+            }, 25, 25, TimeUnit.SECONDS);
+
+            this.session = session;
+        }
+    }
+
+    @Override
+    public void onWebSocketError(@NonNullByDefault({}) Throwable cause) {
+        Throwable localThrowable = (cause != null) ? cause
+                : new IllegalStateException("Null Exception passed to onWebSocketError");
+        logger.debug("Error during websocket communication: {}", localThrowable.getMessage());
+
+        Session localSession = session;
+        if (localSession != null) {
+            stopKeepAlive();
+            localSession.close(StatusCode.SERVER_ERROR, "Failure: " + localThrowable.getMessage());
+            session = null;
+        }
+
+        try {
+            for (ZwaveEventListener listener : listeners) {
+                listener.onConnectionError(String.format("Error: %s", localThrowable.getMessage()));
+            }
+        } catch (Exception e) {
+            logger.warn("Error invoking event listener on error", e);
+        }
+    }
+
+    @Override
+    public void onWebSocketBinary(@NonNullByDefault({}) byte[] payload, int offset, int len) {
+        throw new UnsupportedOperationException("Unimplemented method 'onWebSocketBinary'");
+    }
+
+    @Override
+    public void onWebSocketText(@NonNullByDefault({}) String message) {
+        if (message.contains("\"event\":\"statistics updated\"")) {
+            return;
+        }
+
+        BaseMessage baseEvent = Objects.requireNonNull(gson.fromJson(message, BaseMessage.class));
+
+        if (baseEvent.type == null) {
+            logger.warn("Event with unknown type received.");
+            logger.trace("DATA >> {}", message);
```

Comment:
Minor request: You use "DATA >>" for logging both sent and received messages. => Instead could you please use "DATA >>" for sent and "DATA <<" for received? .. Or something like "SEND >>" resp. "RECV <<"